src/createrepo_c.c

```diff
@@ -59,6 +59,16 @@
 #define OUTDELTADIR "drpms/"
 #define DEFAULT_DATABASE_VERSION    10
 
+/*
+ * Starting with glib 2.70.0, g_pattern_spec_match() replaces
+ * g_pattern_match().
+ */
+#ifdef GLIB_VERSION_2_70
```

Comment:
OK so I've been pointed to this issue. For future reference, a better way to do this would be `#if GLIB_CHECK_VERSION(2, 70, 0)`. The macro you used is intended for declaring API availability; for checking the GLib version in use, `GLIB_CHECK_VERSION` is the easiest way, but you can also use `GLIB_MAJOR_VERSION`, `GLIB_MINOR_VERSION`, and `GLIB_MICRO_VERSION`.

That said, I will change the RHEL 9 patch to drop use of `GLIB_VERSION_2_70` anyway since it's easy to do so. I had done it this way because I thought "we want to show the API as new in version 2.70 so that anybody using the RHEL 9 version of GLib doesn't think the API is present in upstream version 2.68," but it's possible to use `GLIB_VERSION_2_68` when declaring the API yet still show it as 2.70 in the documentation.

Reply:
Actually I think I will leave things be, so that the GPowerProfileMonitor APIs remain treated as version 2.70 for the purposes of `GLIB_VERSION_MIN_REQUIRED` and `GLIB_VERSION_MAX_ALLOWED`. I think this should not be a problem for createrepo_c as long as you switch to using `GLIB_CHECK_VERSION`.